.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+name: infoware CI
+on:
+  push:
+  pull_request:
+
+jobs:
+  linux-x64:
+    name: Linux x64 (${{matrix.cfg.cpp-version}})
+    runs-on: ${{matrix.cfg.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          - { os: ubuntu-18.04, cpp-version: g++-8 }
+          - { os: ubuntu-18.04, cpp-version: g++-9 }
+          - { os: ubuntu-18.04, cpp-version: clang++-7 }
+          - { os: ubuntu-18.04, cpp-version: clang++-8 }
+          - { os: ubuntu-18.04, cpp-version: clang++-9 }
+
+    steps:
+      - name: Checkout infoware
+        uses: actions/checkout@v2
+        with:
+          path: main
+
+      - name: Install apt packages
+        run: sudo apt update && sudo apt install -y cmake ${{ matrix.cfg.cpp-version }} ninja-build ocl-icd-opencl-dev
+
+      - name: Generate CMake
+        run: mkdir main/build && cd main/build && cmake -DCMAKE_BUILD_TYPE=Release -DINFOWARE_USE_OPENCL=ON -DINFOWARE_EXAMPLES=ON -DINFOWARE_TESTS=ON -G Ninja ..
+        env:
+          CXX: ${{matrix.cfg.cpp-version}}
+
+      - name: Build Project
+        run: cd main/build && ninja
+
+      - name: Run Tests
+        run: cd main/build && ctest
+
+  macos:
+    name: macOS x64 ${{matrix.cfg.name}}
+    runs-on: ${{matrix.cfg.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        cfg:
+          - { os: macos-10.15,  name: "Catalina", dist: true}
+          - { os: macos-11,     name: "Big Sur",  dist: false}
+
+    steps:
+      - name: Checkout infoware
+        uses: actions/checkout@v2
+        with:
+          path: main
+
+      - name: Install homebrew packages
+        run: brew install cmake ninja openssl
+
+      - name: Generate CMake
+        run: mkdir main/build && cd main/build && cmake -DCMAKE_BUILD_TYPE=Release -DINFOWARE_USE_OPENCL=ON -DINFOWARE_EXAMPLES=ON -DINFOWARE_TESTS=ON -G Ninja ..
+
+      - name: Build Project
+        run: cd main/build && ninja
+
+      - name: Run Tests
+        run: cd main/build && ctest
+
+  windows: # Windows x64 and x86 build matrix
```

Comment:
x64 *is* one of the three x86 arches. Except where strictly needed (the `cfg.arch` field?), please standardise on "i386"

Reply:
Thanks for pointing out the technicality. I've thought x64 as a 64bit OS, and x86 as a 32bit OS. Switched where appropriate to `i386` and `x64`.